src/index.ts

```diff
@@ -78,41 +78,49 @@ export default function wrapMotoko(compiler: Compiler) {
     };
 
     // Function signatures for `mo.parseMotokoTyped()`
+    /* old API, doesn't support cache and recovery */
     function parseMotokoTyped(paths: string): ParseMotokoTypedResult;
     function parseMotokoTyped(paths: string[]): ParseMotokoTypedResult[];
+    /* new API: */
     function parseMotokoTyped(
         paths: string,
         scopeCache: ScopeCache,
+        enable_recovery?: boolean,
     ): [ParseMotokoTypedResult, ScopeCache];
     function parseMotokoTyped(
         paths: string[],
         scopeCache: ScopeCache,
+        enable_recovery?: boolean,
     ): [ParseMotokoTypedResult[], ScopeCache];
     function parseMotokoTyped(
         paths: string | string[],
         scopeCache?: ScopeCache,
+        enable_recovery?: boolean,
     ): [ParseMotokoTypedResult | ParseMotokoTypedResult[], ScopeCache]
         | ParseMotokoTypedResult
         | ParseMotokoTypedResult[] {
         // Handle the case when the cache is not passed for backwards
         // compatibility and return the program alone to keep the old behavior.
+        if (enable_recovery === undefined) {
+            enable_recovery = false;
+        }
         if (arguments.length === 1) {
```

Comment:
Perhaps need to adjust this? `arguments.length` is `0` if only `paths`, `1` if `paths` and `scopeCache`, and `2` if all.

Reply:
I was confused, nothing needs to be done. Ignore this comment.